EIPS/eip-2456.md

```diff
@@ -0,0 +1,166 @@
+
+## Motivation
+
+<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->
+
+For the Muir Glacier and Istanbul upgrades there were a total of at least 4 times when the planned
+upgrade occurred well away from the intended date. For Ropsten the Istanbul upgrade occurred earlier
+because of the sudden addition of hashing power, and for Muir Glacier it happened much later because
+of the removal of that hashing power. For Mainnet the Istanbul transition was days later and the
+Muir Glacier transition was days sooner because of the unpredictable impact of the Difficulty Bomb.
+
+Such unpredictability is bad for node operators such as exchanges and block explorers. Particularly
+an earlier fork may disrupt their maintenance plans. A much shorter window of unpredictability would
+benefit node operators and developers monitoring the upgrade.
+
+## Specification
+
+A 'Transition Eligible Block' is defined as a block where the number of blocks preceding the block
+is evenly divisible by the `TRANSITION_INCREMENT`, i.e.
+`(block.number % TRANSITION_INCREMENT) == 0`. For Mainnet the `TRANSITION_INCREMENT` will be 1000.
+For testnets a `TRANSITION_INCREMENT` of 1024 is recommended.
+
+A specific `TRANSITION_TIME` will be chosen for the network upgrade to activate.
+
+An upgrade will activate at a Transition Eligible Block if all of the following are true:
+
+- The upgrade has not activated already.
+- The timestamp of the block is on or after the `TRANSITION_TIME`.
+- The previous Transition Eligible Block was on or after the `TRANSITION_TIME`.
+
+Once activated the upgrade remains activated for all future blocks.
+
+If multiple upgrades would activate on a Transition Eligible Block all such transitions will
+activate. Note that most network upgrades are defined as "previous upgrade + new rules" so the
+effect may be that only the most recent network upgrade would appear to activate.
+
+After the upgrade transition completes the Hard Fork Coordinator will update the Meta-EIP to note
+what the transition block was. Clients may incorporate this block into their genesis files as an aid
+to synchronization. This transition block number is only advisory and the time base transition rules
+have precedence.
+
+## Rationale
+
+Previously when determining upgrade transitions a specific block number was chosen. This number was
+typically a multiple of 1000. For testnets a practice developed of making the numbers decimal
+palindromes. Retaining the mainnet tradition is easy, but specifying a palindrome algorithm for
+testnets seemed excessive. As a compromise round decimal numbers (ending in `000`) are proposed for
+mainnet and round hexadecimal numbers (ending in `0x000`, `0x400`, `0x800`, or `0xc00`) is proposed
+for testnets.
```

Comment:
This seems unnecessarily complex.  The specific block number choice is pretty arbitrary in all cases, and having a different strategy for mainnet and testnet seems like a waste of engineering effort for what is essentially the color of the boathouse being different in mainnet and testnets.